### PR TITLE
Social integration

### PR DESCRIPTION
**Pull request by danacebryantalfonso**

### CODE REVIEW

File: SocialIntegration/app/src/main/java/com/example/socialIntegration/view/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        }
+    }
+    private fun signIn() {
+        val intent = mGoogleSignInClient!!.signInIntent
+        startActivityForResult(intent, RC_SIGN_IN)
+    }
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        callbackManager?.onActivityResult(requestCode, resultCode, data)
+        super.onActivityResult(requestCode, resultCode, data)
+
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/example/socialIntegration/view/MainActivity.kt:96]: remove extra space

File: SocialIntegration/app/src/main/java/com/example/socialIntegration/view/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        try {
+            //val account = completedTask.getResult(ApiException::class.java)
+            launchHome()
+            // Signed in successfully, show authenticated UI.
+            //updateUI(account)
+        } catch (e: ApiException) {
+            // The ApiException status code indicates the detailed failure reason.
+            // Please refer to the GoogleSignInStatusCodes class reference for more information.
+            println("signInResult:failed code=" + e.statusCode)
+            //updateUI(null)
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/example/socialIntegration/view/MainActivity.kt:122]: remove unnecessary comments
use /**/ for more than 1 comment

File: SocialIntegration/app/src/main/java/com/example/socialIntegration/view/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        }
+    }
+
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/example/socialIntegration/view/MainActivity.kt:125]: remove extra line

File: SocialIntegration/app/src/main/java/com/example/socialIntegration/view/ui/gallery/GalleryFragment.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        galleryViewModel =
+            ViewModelProvider(this).get(GalleryViewModel::class.java)
+        val root = inflater.inflate(R.layout.fragment_gallery, container, false)
+        val textView: TextView = root.findViewById(R.id.text_gallery)
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/example/socialIntegration/view/ui/gallery/GalleryFragment.kt]: use proper naming of variables, pls check other variables

File: SocialIntegration/app/src/main/java/com/example/socialIntegration/viewmodel/ListViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                    val request = GraphRequest.newMeRequest(
+                        accessToken
+                    ) { `object`, response ->
+                        // Application code
+
+                        println(`object`.toString())
+                        println(`object`.toString())
+                        println(`object`.toString())
+                        println(`object`.toString())
+                        println(`object`.toString())
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/java/com/example/socialIntegration/viewmodel/ListViewModel.kt]: if not use, pls delete it

File: SocialIntegration/app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                android:layout_width="213dp"
+                android:layout_height="66dp"
+                android:text="@string/btnSignUp"
+                tools:layout_editor_absoluteX="42dp"
+                tools:layout_editor_absoluteY="231dp"
+                tools:ignore="MissingConstraints" />
+
+            <Button
+                android:id="@+id/button2"
+                android:layout_width="wrap_content"
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/res/layout/activity_main.xml]: proper naming of button (android:id)

File: SocialIntegration/app/src/main/res/navigation/nav_graph.xml [hunk 7]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<navigation xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/nav_graph"
+    app:startDestination="@id/FirstFragment">
+
+    <fragment
+        android:id="@+id/FirstFragment"
```
Comment (rexendy-lig) [inline SocialIntegration/app/src/main/res/navigation/nav_graph.xml]: use snake case

**Review by rexendy-lig — CHANGES_REQUESTED** [review]

File: SocialIntegration/app/build.gradle [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+}
+
+dependencies {
+    implementation 'androidx.navigation:navigation-fragment:2.3.4'
+    implementation 'androidx.navigation:navigation-ui:2.3.4'
```
[the remark this answers is not in the document]
Comment (danacebryantalfonso) [reply SocialIntegration/app/build.gradle]: done in 46233fbe41558cfda6a8607d80bb1cce71f1e1e1

File: SocialIntegration/app/src/main/java/com/example/socialIntegration/view/HomeActivity.kt [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        val fab: FloatingActionButton = findViewById(R.id.fab)
+        fab.setOnClickListener { view ->
+            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
+                .setAction("Action", null).show()
+        }
+        val drawerLayout: DrawerLayout = findViewById(R.id.drawer_layout)
+        val navView: NavigationView = findViewById(R.id.nav_view)
+        val navController = findNavController(R.id.nav_host_fragment)
+        // Passing each menu ID as a set of Ids because each
+        // menu should be considered as top level destinations.
```
[the remark this answers is not in the document]
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/example/socialIntegration/view/HomeActivity.kt]: revised in 46233fbe41558cfda6a8607d80bb1cce71f1e1e1

File: SocialIntegration/app/src/main/java/com/example/socialIntegration/view/HomeActivity.kt [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+        appBarConfiguration = AppBarConfiguration(
+                setOf(
+                        R.id.nav_home, R.id.nav_gallery, R.id.nav_slideshow
+                ), drawerLayout
```
[the remark this answers is not in the document]
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/example/socialIntegration/view/HomeActivity.kt]: revised in 46233fbe41558cfda6a8607d80bb1cce71f1e1e1

File: SocialIntegration/app/src/main/java/com/example/socialIntegration/view/MainActivity.kt [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+    private var mGoogleSignInClient: GoogleSignInClient? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        googleSignIn()
+        facebookSignIn()
+    }
+    private fun googleSignIn() {
```
[the remark this answers is not in the document]
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/example/socialIntegration/view/MainActivity.kt]: revised in 46233fbe41558cfda6a8607d80bb1cce71f1e1e1

File: SocialIntegration/app/src/main/java/com/example/socialIntegration/view/MainActivity.kt [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+        callbackManager = CallbackManager.Factory.create()
+        LoginManager.getInstance().registerCallback(callbackManager,
+            object : FacebookCallback<LoginResult?> {
+                @SuppressLint("WrongConstant")
+                val accessToken = AccessToken.getCurrentAccessToken()
+                val isLoggedIn = accessToken != null && !accessToken.isExpired
+
+                override fun onSuccess(loginResult: LoginResult?) {
+
+
```
[the remark this answers is not in the document]
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/example/socialIntegration/view/MainActivity.kt]: revised in 46233fbe41558cfda6a8607d80bb1cce71f1e1e1

File: SocialIntegration/app/src/main/java/com/example/socialIntegration/view/MainActivity.kt (hunk 3 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/example/socialIntegration/view/MainActivity.kt]: revised in 46233fbe41558cfda6a8607d80bb1cce71f1e1e1

File: SocialIntegration/app/src/main/java/com/example/socialIntegration/view/ui/gallery/GalleryFragment.kt (hunk 4 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/example/socialIntegration/view/ui/gallery/GalleryFragment.kt]: revised in 46233fbe41558cfda6a8607d80bb1cce71f1e1e1

File: SocialIntegration/app/src/main/java/com/example/socialIntegration/viewmodel/ListViewModel.kt (hunk 5 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/java/com/example/socialIntegration/viewmodel/ListViewModel.kt]: revised in 46233fbe41558cfda6a8607d80bb1cce71f1e1e1

File: SocialIntegration/app/src/main/res/layout/activity_main.xml (hunk 6 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/res/layout/activity_main.xml]: revised in 46233fbe41558cfda6a8607d80bb1cce71f1e1e1

File: SocialIntegration/app/src/main/res/navigation/nav_graph.xml (hunk 7 above)
Comment (danacebryantalfonso) [reply SocialIntegration/app/src/main/res/navigation/nav_graph.xml]: revised in 46233fbe41558cfda6a8607d80bb1cce71f1e1e1

**Review by rexendy-lig — APPROVED** [review]

**Pull request merged**